pkg/gce-pd-csi-driver/controller_test.go

```diff
@@ -1413,3 +1413,76 @@ func TestPickRandAndConsecutive(t *testing.T) {
 
 	}
 }
+
+func TestVolumeOperationConcurrency(t *testing.T) {
+	snapshotToExecute := make(chan gce.SnapshotSourceAndTarget, 3)
+	readyToExecute := make(chan struct{}, 2)
+	reqFinished := make(chan error, 2)
+
+	gceDriver := initBlockingGCEDriver(t, nil, snapshotToExecute, readyToExecute)
+	cs := gceDriver.cs
+
+	vol1CreateSnapshotReqA := &csi.CreateSnapshotRequest{
+		Name:           name + "1A",
+		SourceVolumeId: testVolumeId + "1",
+	}
+	vol1CreateSnapshotReqB := &csi.CreateSnapshotRequest{
+		Name:           name + "1B",
+		SourceVolumeId: testVolumeId + "1",
+	}
+	vol2CreateSnapshotReq := &csi.CreateSnapshotRequest{
+		Name:           name + "2",
+		SourceVolumeId: testVolumeId + "2",
+	}
+
+	runRequestInBackground := func(req *csi.CreateSnapshotRequest) {
```

Comment:
I'd drop the "InBackground" from this func name. Prefer to name things after what they are or what they do, rather than name them after how they are used.